openassessment/xblock/staff_area_mixin.py

```diff
@@ -378,18 +378,49 @@ def get_student_info_path_and_context(self, student_username, expanded_view=None
         self_assessment = None
         peer_assessments = None
         submitted_assessments = None
+        peer_assessments_grade_context = []
+        self_assessment_grade_context = None
+        example_based_assessment_grade_context = None
+        staff_assessment_grade_context = None
+        grade_details = None
+
+        workflow = self.get_workflow_info(submission_uuid=submission_uuid)
+        grade_exists = workflow.get('status') == "done"
 
         if "peer-assessment" in assessment_steps:
             peer_assessments = peer_api.get_assessments(submission_uuid)
             submitted_assessments = peer_api.get_submitted_assessments(submission_uuid)
 
+            if grade_exists:
+                peer_api.get_score(submission_uuid, self.workflow_requirements()["peer"])
+                peer_assessments_grade_context = [
+                    self._assessment_grade_context(peer_assessment)
+                    for peer_assessment in peer_assessments
+                ]
+
         if "self-assessment" in assessment_steps:
             self_assessment = self_api.get_assessment(submission_uuid)
+            if grade_exists:
+                self_assessment_grade_context = self._assessment_grade_context(self_assessment)
 
         if "example-based-assessment" in assessment_steps:
             example_based_assessment = ai_api.get_latest_assessment(submission_uuid)
-
-        workflow = self.get_workflow_info(submission_uuid=submission_uuid)
+            if grade_exists:
+                example_based_assessment_grade_context = self._assessment_grade_context(example_based_assessment)
+
+        if grade_exists:
+            staff_assessment = staff_api.get_latest_staff_assessment(submission_uuid)
```

Comment:
I don't think `staff_assessment` is defined above, we'll need to coordinate this with my in-flight PR once one of them merges.